share/dkg/pedersen/dkg.go

```diff
@@ -343,6 +380,32 @@ func (d *DistKeyGenerator) DistKeyShare() (*DistKeyShare, error) {
 	}, nil
 }
 
+
+func (f *DistKeyShare) Renew(g *DistKeyShare, suite Suite) (*DistKeyShare, error){
+	//Check G(0) = 0*G.
+	if !g.Public().Equal(suite.Point().Base().Mul(suite.Scalar().Zero(),nil)){
+		return nil, errors.New("G(0) != 0*G, wrong renewal function!")
```

Comment:
I think go lint will tell you but avoid "!" at the end of your errors, stick to simple error sentences that can be embedded in larger error report etc...